lib/child_process.js

```diff
@@ -183,8 +179,8 @@ exports.execFile = function(file /*, args, options, callback*/) {
       stdout = Buffer.concat(_stdout);
       stderr = Buffer.concat(_stderr);
     } else {
-      stdout = _stdout;
-      stderr = _stderr;
+      stdout = Buffer.concat(_stdout).toString(encoding);
```

Comment:
As an easy optimization we should track the number of received bytes for each stream and pass that to `Buffer.concat()`.